data-prepper-plugins/elasticsearch/src/main/resources/otel-v1-apm-span-index-template.json

```diff
@@ -40,8 +40,21 @@
         "type": "keyword"
       },
       "traceGroup": {
-        "ignore_above": 1024,
-        "type": "keyword"
+        "properties": {
```

Comment:
This will break existing index data and kibana plugin. Did we test it?

Reply:
Good question - just built and tested, ran into this:

>2021-04-14T20:06:55,580 [raw-pipeline-process-worker-3-thread-2] WARN  com.amazon.dataprepper.plugins.sink.elasticsearch.ElasticsearchSink - Document [index {[otel-v1-apm-span][_doc][5cc8fdea8bda115f], source[{"traceId":"44bbcc035e9b60f4b321d75d341e3ebe","spanId":"5cc8fdea8bda115f","traceState":"","parentSpanId":"1d5fe371e312dbd3","name":"HTTP POST","kind":"SPAN_KIND_CLIENT","startTime":"2021-04-14T20:06:20.896851712Z","endTime":"2021-04-14T20:06:20.901840128Z","durationInNanos":4988416,"serviceName":"order","events":[],"links":[],"droppedAttributesCount":0,"droppedEventsCount":0,"droppedLinksCount":0,"traceGroup.name":"client_delivery_status","traceGroup.endTime":"2021-04-14T20:06:20.913838336Z","traceGroup.statusCode":0,"traceGroup.durationInNanos":59541504,"span.attributes.http@url":"http://localhost:8087/logs","instrumentationLibrary.version":"0.14b0","resource.attributes.telemetry@sdk@language":"python","span.attributes.http@status_text":"","resource.attributes.telemetry@sdk@version":"0.14b0","resource.attributes.service@instance@id":"140366117791112","resource.attributes.service@name":"order","span.attributes.component":"http","status.code":0,"instrumentationLibrary.name":"opentelemetry.instrumentation.requests","span.attributes.http@method":"POST","resource.attributes.telemetry@sdk@name":"opentelemetry","resource.attributes.host@hostname":"xxxxxx.ec2.internal","span.attributes.http@status_code":200}]}] has failure: ElasticsearchException[Elasticsearch exception [type=mapper_parsing_exception, reason=Could not dynamically add mapping for field [traceGroup.name]. Existing mapping for [traceGroup] must be of type object but found [keyword].]]

The data never reaches the cluster, it fails to be written by the ES sink. So this is a breaking change - any solutions to this other than requiring users to clear their data?

Reply:
This is because the ES index template has changed. Earlier TraceGroup was keyword, now its not. This is why we should maybe consider them as separate fields.

Reply:
Agreed. That may be the only option to support backward compatibility. Maybe we can do field names as such
```
traceGroup (name)
traceGroup@statusCode
traceGroup@endTime
traceGroup@durationInNanos
```
that way we do not lose the compatibility. Alternatively, we use a different prefix e.g.
```
traceGroupInfo.name
traceGroupInfo.statusCode
traceGroupInfo.endTime
traceGroupInfo.durationInNanos
```
But then one has to hardcode the population of both `traceGroup` and `traceGroupInfo.name` to support backward compatibility, which I do not think is a good design. Can discuss further offline.

Reply:
BTW, I tested that data-ingestion works fine with the sample app (given that no old data is in the ODFE). But trace-analytics-ui breaks as expected.

Reply:
Yeah. Can we actually keep `traceGroup` as key word and add new property called `traceGroupFields`. Also, you should check with @joshuali925 .

Reply:
Make the change and the compatibility issue is resolved by combining together with https://github.com/opendistro-for-elasticsearch/trace-analytics/pull/33